docs/feature-requests/006-pull-request-reviewer-flow.md

```diff
@@ -0,0 +1,156 @@
+**_Part 1 - Required information_**
+
+# Pull Request Review -- Reviewer Flow
+
+## :memo: Summary
+
+Provide code review to an existing pull request within Atom.
+
+*Note*: This RFC is an iteration of the [original RFC for Pull Request Review](./003-pull-request-review.md).
+
+## :checkered_flag: Motivation
+
+We already have an innovative review-comments-in-dock (RCID) workflow built out for the receiving end of pull request reviews. In order to complete the full code review experience within Atom, we should also build out a workflow for users to author pull request reviews.
+
+## 🤯 Workflow Explanation
+
+This is a high level overview of what the workflow of a PR Review author should look like. More on the functionality and behaviour of each component in the next section.
+
+#### 1. Start a review
+
+There are three ways to start a review:
+1. Click "Start a review" button on the header of review dock, or footer of PR detail item, or on the empty state of review dock
+2. [Respond to an existing review thread by clicking "Start a review"](#responding-to-a-comment-thread)
+3. [Click on a "add comment" icon on the gutter](#add-comment-gutter-icon)
+
+#### 2. Continue a review
+
+Once a pending review has been started, user can add more comments to it by:
+1. [Responding to an existing review thread](#responding-to-a-comment-thread)
+2. [Clicking "add comment" icon on the gutter](#add-comment-gutter-icon)
+
+#### 3. Submit a review
+The only way to submit a review within Atom is by using the ["Submit review" button in the Pending Review tab](#summary-section). After publishing the review, the Pending Review tab will be destroyed. User will be led back to the All Reviews tab, which will immediately reflect the just published review.
+
+## 🤯 Components Explanation
+
+
+### "All Reviews" tab
+This tab shows all review summaries and review comments, including the ones that are part of a _pending review_ that has not been submitted yet.
```

Comment:
Hmm... since the pending review will be in the "All Reviews" tab, is it necessary to have a separate "Pending Review" tab?

Quite possibly. I haven't thought too deeply about it. Just wanted to throw out the question.

Reply:
My major concerns with keeping both published comments and pending comments in the same space are that 1) it would get too crowded, and 2) there would be no obvious hierarchy in the information presented. 

I would argue that when a user is in the middle of reviewing a PR (i.e. have a review pending), the most relevant information to them would be their pending comments, as well as the summary textbox that they need to fill out before publishing a review. I couldn't figure out a way to fit all this along with other published comments, while preserving the hierarchy of importance.